src/logic.ts

```diff
@@ -0,0 +1,16 @@
+import get from 'lodash/get'
+import has from 'lodash/has'
+
+export const when = (check: (ctx) => Promise<Boolean> | Boolean, middleware: (ctx, next) => Promise<any>) => {
+  return async (ctx, next) => {
+    const proceed = await check(ctx)
+    if(proceed) {
```

Comment:
if we are expecting it to return `Boolean`, shall we restrict to boolean type?
eg:
```js
assert(isBoolean(proceed))
if(proceed) {
} else {}
```

if not, might as well accept any truthy value such as 
```js
check: (ctx) => Promise<Any> | Any
```

Reply:
I think you're confusing the Typescript side and the Javascript side of things.

Typing it to `any` doesn't help people understand what is expected. We expect it to be a boolean and they should follow that. But, *if* someone decides they want to go against the expectation and pass a truthy value, it's on them and they will have to cast that value. But I'm not stopping them by raising an exception

Reply:
Well, is just a suggestion. 

if anyone uses this package without typescript, `type` checking will not be available, thus passing truthy value will be sufficient.